src/screens/WalletType.js

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import logo from '../assets/images/hathor-logo.png';
+import wallet from '../utils/wallet';
+import ledger from '../utils/ledger';
+import hathorLib from '@hathor/wallet-lib';
+import { IPC_RENDERER } from '../constants';
+import InitialImages from '../components/InitialImages';
+
+
+/**
+ * Screen used to select between hardware wallet or software wallet
+ *
+ * @memberof Screens
+ */
+class WalletType extends React.Component {
+  // Trial parameters when trying to connect to ledger
+  // trialNumber: current trial
+  // trialLimit: maximum number of trials
+  // trialInterval: wait interval between trials
+  trialNumber = 0
+  trialLimit = 100
+  trialInterval = 3000
+
+  /**
+   * errorMessage {string} message to show in case of error connecting to ledger
+   * hardware {boolean} if hardware wallet was chosen
+   * waitAction {boolean} after connecting to ledger we need to wait for user action
+   */
+  state = {
+    errorMessage: '',
+    hardware: false,
+    waitAction: false,
+  }
+
+  componentDidMount() {
+    // Update Sentry when user started wallet now
+    wallet.updateSentryState();
+
+    if (IPC_RENDERER) {
+      IPC_RENDERER.on("ledger:version", (event, arg) => {
+        if (arg.success) {
+          // We wait 2 seconds to update the message on the screen
+          // so the user don't see the screen updating fast
+          setTimeout(() => {
+            this.setState({ waitAction: true, errorMessage: '' }, () => {
+              ledger.getPublicKeyData();
+            })
+          }, 2000);
+        } else {
+          if (!this.state.hardware) {
+            // User clicked 'Back' button
+            this.trialNumber = 0;
+            return;
+          }
+
+          if (this.trialNumber < this.trialLimit) {
+            setTimeout(() => {
+              this.trialNumber += 1;
+              ledger.getVersion();
+            }, this.trialInterval);
+          } else {
+            // Error
+            this.setState({ errorMessage: arg.error });
+          }
+        }
+      });
+
+      IPC_RENDERER.on("ledger:publicKeyData", (event, arg) => {
+        if (!this.state.hardware) {
```

Comment:
Maybe we should refactor and move this code to a method in the class.

Reply:
Done